### PR TITLE
PYT-45: todolist contract

### PR DESCRIPTION
**Pull request by abhishekrapid**
Please review the code.

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/** @type import('hardhat/config').HardhatUserConfig */
+require("@nomicfoundation/hardhat-chai-matchers");
+
+// Go to https://www.alchemyapi.io, sign up, create
+// a new App in its dashboard, and replace "KEY" with its key
+const ALCHEMY_API_KEY = "KEY";
```
Comment (apoorv-nag) [inline hardhat.config.js:6]: take key from env file using nodejs @abhishekrapid

File: hardhat.config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+// Replace this private key with your Goerli account private key
+// To export your private key from Metamask, open Metamask and
+// go to Account Details > Export Private Key
+// Beware: NEVER put real Ether into testing accounts
+const GOERLI_PRIVATE_KEY = "YOUR GOERLI PRIVATE KEY";
```
Comment (apoorv-nag) [inline hardhat.config.js:12]: take key from env file using nodejs @abhishekrapid

File: contracts/todolist.sol [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        view
+        onlyOwner
+        returns (string memory)
+    {
+        return _listOfTodos[taskId_].title;
+    }
+
+    // Removes a task from the list
+    function removeTask(uint256 taskId_) external onlyOwner nonReentrant{
+        require(taskId_ <= _listOfTodos.length, "Tasks are less");
```
Comment (apoorv-nag) [inline contracts/todolist.sol:77]: use custom error which is a new feature of solidity 0.8 @abhishekrapid

File: contracts/todolist.sol [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        return _listOfTodos[taskId_].status;
+    }
+
+    function getTaskTextAtIndex(uint taskId_)
+        external
+        view
+        onlyOwner
+        returns (string memory)
+    {
+        return _listOfTodos[taskId_].title;
```
Comment (apoorv-nag) [inline contracts/todolist.sol:72]: check if id do not exist @abhishekrapid

File: contracts/todolist.sol [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        _listOfTodos[taskId_].status = TodoStatuses.COMPLETE;
+    }
+
+    function getTaskStatusAtIndex(uint taskId_)
+        external
+        view
+        onlyOwner
+        returns (TodoStatuses)
+    {
+        return _listOfTodos[taskId_].status;
```
Comment (apoorv-nag) [inline contracts/todolist.sol:63]: check if id do not exist @abhishekrapid

**Review by apoorv-nag — CHANGES_REQUESTED** [review]

File: hardhat.config.js (hunk 1 above)
Comment (abhishekrapid) [reply hardhat.config.js]: Done

File: hardhat.config.js (hunk 2 above)
Comment (abhishekrapid) [reply hardhat.config.js]: Done

File: contracts/todolist.sol (hunk 3 above)
Comment (abhishekrapid) [reply contracts/todolist.sol]: Done

File: contracts/todolist.sol (hunk 4 above)
Comment (abhishekrapid) [reply contracts/todolist.sol:86]: Done

File: contracts/todolist.sol [hunk 6]
```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;  // Specifies the version of solidity that our code is written with
```
Comment (apoorv-nag) [inline contracts/todolist.sol:3]: Solidity version not matching wtih hardhat config.

**Review by apoorv-nag — CHANGES_REQUESTED** [review]

**Comment by apoorv-nag** [comment]
@abhishekrapid please fill


**Description**


**Related Issue**





**Motivation and Context**


**How Has This Been Tested?**




**Screenshots (if appropriate):**

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**


- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

**Pull request merged**